lms/djangoapps/instructor_task/tests/test_tasks_helper.py

```diff
@@ -1791,22 +1792,16 @@ def test_grade_report(self):
                 ignore_other_columns=True,
             )
 
-    def test_fast_generation_zero_grade(self):
+    @ddt.data(True, False)
+    def test_fast_generation(self, create_non_zero_grade):
+        if create_non_zero_grade:
+            self.submit_student_answer(self.student.username, u'Problem1', ['Option 1'])
         with patch('lms.djangoapps.instructor_task.tasks_helper.runner._get_current_task'):
-            with patch('lms.djangoapps.grades.course_grade.CourseGradeBase._prep_course_for_grading') as mock_grader:
+            with patch('lms.djangoapps.grades.course_data.get_course_blocks') as mock_course_blocks:
                 with patch('lms.djangoapps.grades.subsection_grade.get_score') as mock_get_score:
                     CourseGradeReport.generate(None, None, self.course.id, None, 'graded')
-                    self.assertFalse(mock_grader.called)
                     self.assertFalse(mock_get_score.called)
-
-    def test_slow_generation_nonzero_grade(self):
-        self.submit_student_answer(self.student.username, u'Problem1', ['Option 1'])
-        with patch('lms.djangoapps.instructor_task.tasks_helper.runner._get_current_task'):
-            with patch('lms.djangoapps.grades.course_grade.CourseGradeBase._prep_course_for_grading') as mock_grader:
-                with patch('lms.djangoapps.grades.subsection_grade.get_score') as mock_get_score:
-                    CourseGradeReport.generate(None, None, self.course.id, None, 'graded')
-                    self.assertTrue(mock_grader.called)
-                    self.assertTrue(mock_get_score.called)
+                    self.assertFalse(mock_course_blocks.called)
```

Comment:
**Note to reviewers:** This test verifies that `get_course_blocks` (user-specific block-structure transforms) and `get_score` (problem-score retrieval from CSM/Submissions) don't happen when `CourseGradeReport` is generated.